### PR TITLE
[PLT-6627] Unit Tests for Badge

### PR DESCRIPTION
**Pull request by imogenem**
## Description
Unit tests for Badge 

## Notes
Found a spelling error in the docs markdown file 

## Where to Start
Badge.test.jsx in the riipen-ui components folder
badge.md in the riipen-ui-docs components folder

### CODE REVIEW

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+      const component = "abcdefg";
+
+      mount(<Badge component={component} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("content prop", () => {
+    it("badge does not render when content is null and showZero is false", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:151]: I think a better test would be "when content is 0". `null` will never render, but 0 will render if `showZero` is `true`.

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+    it("badge renders with content and showZero props", () => {
+      const content = 2;
+      const showZero = true;
+
+      const wrapper = mount(<Badge content={content} showZero={showZero} />);
+
+      expect(Number(wrapper.find("Badge").text())).toEqual(2);
+    });
+  });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:172]: I think we should add one more test here. "It should render truthy content" where content is just set to 1 or 2 or something. You covered the other two edge cases very well, but I think we need this base case as well.

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+  describe("max prop", () => {
+    it("displays max value given max greater than content value", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:175]: Should this say "displays max value given content greater than max value"?

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+      const max = 9;
+      const content = 11;
+
+      const wrapper = mount(<Badge max={max} content={content} />);
+
+      expect(wrapper.text()).toEqual(`${max}+`);
+    });
+
+    it("displays content value given content greater than max value", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:184]: I think this should say "display content value given content less than max value".

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass(size)
+      ).toEqual(true);
+    });
+
+    it("does not set size class name when invalid variant prop used", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:251]: Instead of "invalid variant" I would be tempted to say "non standard". This makes it more clear to anyone reading the test that invalid actually means non standard.

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+          .childAt(0)
+          .hasClass(variant)
+      ).toEqual(true);
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("size")
+      ).toEqual(false);
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:295]: I think we can remove this as we already tested it on line 257 above.

**Review by jordanell — CHANGES_REQUESTED** [review]
Really nice job! Mostly just small things to clean up.

File: packages/riipen-ui/src/components/Badge.test.jsx (hunk 5 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Badge.test.jsx:251]: Can I revise it to "non standard variant" to convey that it's the variant prop controlling if size can be used?

Comment (jordanell) [reply packages/riipen-ui/src/components/Badge.test.jsx:251]: Love it

File: packages/riipen-ui/src/components/Badge.test.jsx (hunk 1 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Badge.test.jsx:151]: Added the the content prop and adjusted the test name a bit more.

File: packages/riipen-ui/src/components/Badge.test.jsx [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+
+      expect(
+        wrapper
+          .find("Badge")
+          .childAt(0)
+          .childAt(0)
+          .hasClass("hidden")
+      ).toEqual(true);
+      expect(Number(wrapper.text())).toEqual(content);
+    });
```
Comment (jordanell) [inline packages/riipen-ui/src/components/Badge.test.jsx:164]: Slightly confused. According to the component, the "0" should not render if showZero is `false` (which is the default`. so why is line 163 actually working? Maybe it's because it actually _does_ render, but the CSS is hiding it.

I would maybe recommend just deleting line 163 as it's sort of confusing and rename the test maybe to "it hides the content when content is 0 and showZero is false".

Does that make sense?

**Review by jordanell — CHANGES_REQUESTED** [review]
One last little thing.

File: packages/riipen-ui/src/components/Badge.test.jsx (hunk 7 above)
Comment (imogenem) [reply packages/riipen-ui/src/components/Badge.test.jsx:164]: Yes it does

**Review by jordanell — APPROVED** [review]
